app/scripts/ui/Map.js

```diff
@@ -0,0 +1,36 @@
+var React  = require('react');
+var Router = require('react-router');
+
+/** @jsx React.DOM */
+var Map = React.createClass({
+    getDefaultProps: function(){
+        return{
+            initialZoom: 15,
+            centerLat: 12.97160,
+            centerLng: 77.59456
+        };
+    },
+    getMapCenter: function(){
+        return new google.maps.LatLng(this.props.centerLat,this.props.centerLng);
+    },
+    componentDidMount: function(rootNode){
+        var mapSettings = {
+            center: this.getMapCenter(),
+            zoom: this.props.initialZoom
+        };
+        //TODO:: Ivin: Couldnt find a way to pass the dom reference from the place of usage. Hardcoding for now.
```

Comment:
@ivinjose check this http://facebook.github.io/react/docs/working-with-the-browser.html

I made the changes and pushed to this branch. Please refer the code changes